bittensor/_subtensor/__init__.py

```diff
@@ -197,6 +199,10 @@ def add_defaults(cls, defaults ):
         defaults.subtensor.chain_endpoint = os.getenv('BT_SUBTENSOR_CHAIN_ENDPOINT') if os.getenv('BT_SUBTENSOR_CHAIN_ENDPOINT') != None else None
         defaults.subtensor._mock = os.getenv('BT_SUBTENSOR_MOCK') if os.getenv('BT_SUBTENSOR_MOCK') != None else False
 
+        defaults.subtensor.register = bittensor.Config()
+        defaults.subtensor.register.num_processes = os.getenv('BT_SUBTENSOR_REGISTER_NUM_PROCESSES') if os.getenv('BT_SUBTENSOR_REGISTER_NUM_PROCESSES') != None else None # uses processor count by default within the function
```

Comment:
isnt the default from the OS? 
if so we can remove the non check in the registration etc

Reply:
if we pass None as an arg to the register function, it doesn't use a default arg, it gets None as the arg.
So, I do a None check and default to the number of CPUs the OS allows for the process.
I figure it's better to leave this logic (using OS) in the cpu registration as the flag will be used for cuda too, potentially